### DISCUSSION TITLE
Php 8.1 with Phalcon

### DISCUSSION THREAD

**Discussion by euazzo** [phalcon/cphalcon#15898] [General]
I need to create a new docker image to update an application to php 8.1.
However, I encounter a compatibility problem between php 8.1 and the Phalcon extension.
The current configuration is Phalcon 3.4.1 with PHP 7.11.

`FROM php:X.X-apache
MAINTAINER IT Department - Consulcesi Tech SA

ARG PHALCON_VERSION=XXX
ARG PHALCON_EXT_PATH=phpX/64bits
 
RUN set -xe && \
       # Compile Phalcon
        curl -LO https://github.com/phalcon/cphalcon/archive/v${PHALCON_VERSION}.tar.gz && \
        tar xzf ${PWD}/v${PHALCON_VERSION}.tar.gz && \
        docker-php-ext-install -j $(getconf _NPROCESSORS_ONLN) ${PWD}/cphalcon-${PHALCON_VERSION}/build/${PHALCON_EXT_PATH} && \
        # Remove all temp files
        rm -r \
            ${PWD}/v${PHALCON_VERSION}.tar.gz \
            ${PWD}/cphalcon-${PHALCON_VERSION}
`
The error is on the build phase at the Phalcon run section. I also tried with different VERSION and EXT_PATH.
It's possible? What version/configuration should I set to get an image with php8.1 and Phalcon?

**Comment by AzeemOguntola** [comment]
Only Phalcon 5 is compatible with PHP 8. This is currently in beta phase.

**Comment by euazzo** [comment]
What is it the highest PHP version supported by the latest stable version (4.1.3) of Phalcon?
When will the stable version of Pahlcon 5 be released?

**Reply by AzeemOguntola** [reply]
7.4 is the highest version of PHP you can use with Phalcon 4. 

As for Phalcon 5, it is currently v5.0.0beta3. I don't think there is a specific date, i guess as soon as all major defects are fixed.
